kv/kvclient/kvcoord: TestTxnDBDirtyWriteAnomaly failed

kv/kvclient/kvcoord.TestTxnDBDirtyWriteAnomaly [failed](https://tanzanite.cluster.engflow.com/invocations/default/71c00477-716a-4bd4-b702-33e4006834ff?testReportRun=4&testReportShard=9&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZjbGllbnQva3Zjb29yZDprdmNvb3JkX3Rlc3Q=) on master @ [f2ce52ff7cc3b4680e0fec2ef844baad49042d6a](https://github.com/cockroachdb/cockroach/commits/f2ce52ff7cc3b4680e0fec2ef844baad49042d6a):

Fatal error:

```
panic: test timed out after 5m0s
running tests:
	TestTxnDBDirtyWriteAnomaly (1m53s)
	TestTxnDBDirtyWriteAnomaly/both_commit (16s)

```
Stack: 

```
goroutine 33556 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x385
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestTxnDBDirtyWriteAnomaly
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestTxnDBDirtyWriteAnomaly2810566716
    test_log_scope.go:76: use -show-logs to present logs inline
=== RUN   TestTxnDBDirtyWriteAnomaly/both_abort
=== RUN   TestTxnDBDirtyWriteAnomaly/both_commit
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>run=4</code>
 - <code>shard=9</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTxnDBDirtyWriteAnomaly.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-45002

--- a/pkg/kv/kvclient/kvcoord/BUILD.bazel
+++ b/pkg/kv/kvclient/kvcoord/BUILD.bazel
@@ -119,7 +119,7 @@ gomock(
 
 go_test(
     name = "kvcoord_test",
-    size = "medium",
+    size = "large",
     srcs = [
         "batch_test.go",
         "condensable_span_set_test.go",

